cmd/generate-catalog/generate_test.go

```diff
@@ -0,0 +1,685 @@
+	names := []string{"rhacs-3.62", "latest", "rhacs-4.0", "rhacs-4.1", "stable"}
+	for i, ch := range channels {
+		assert.Equal(t, names[i], ch.Name)
+	}
+}
+
+func TestClearReplacesForStartingEntries(t *testing.T) {
+	channels := []Channel{
+		{
+			Name:           "rhacs-3.62",
+			yStreamVersion: semver.MustParse("3.62.0"),
+			Entries: []ChannelEntry{
+				{Name: "rhacs-operator.v3.62.0", Replaces: "rhacs-operator.v3.61.0", version: semver.MustParse("3.62.0")},
+			},
+		},
+		{
+			Name:           "rhacs-4.0",
+			yStreamVersion: semver.MustParse("4.0.0"),
+			Entries: []ChannelEntry{
+				{Name: "rhacs-operator.v4.0.0", Replaces: "rhacs-operator.v3.62.1", version: semver.MustParse("4.0.0")},
+				{Name: "rhacs-operator.v4.0.1", Replaces: "rhacs-operator.v4.0.0", version: semver.MustParse("4.0.1")},
+				{Name: "rhacs-operator.v4.0.2", Replaces: "rhacs-operator.v4.0.1", version: semver.MustParse("4.0.2")},
+			},
+		},
+		{
+			Name:           "rhacs-4.1",
+			yStreamVersion: semver.MustParse("4.1.0"),
+			Entries:        []ChannelEntry{},
+		},
+	}
+
+	adjustStartingEntries(channels)
+
+	assert.Empty(t, channels[0].Entries[0].Replaces)
+
+	assert.Empty(t, channels[1].Entries[0].Replaces)
+	assert.Equal(t, "rhacs-operator.v4.0.0", channels[1].Entries[1].Replaces)
+	assert.Equal(t, "rhacs-operator.v4.0.1", channels[1].Entries[2].Replaces)
+
+	assert.Len(t, channels[2].Entries, 0)
+}
+
+func TestChannelShouldHaveEntry(t *testing.T) {
+	tests := []struct {
+		name     string
+		channel  Channel
+		entry    ChannelEntry
+		expected bool
+	}{
+		{
+			name:     "Entry belongs to channel",
+			channel:  Channel{yStreamVersion: semver.MustParse("4.0.0")},
+			entry:    ChannelEntry{version: semver.MustParse("4.0.1")},
+			expected: true,
+		},
+		{
+			name:     "Entry is exact Y-stream version",
+			channel:  Channel{yStreamVersion: semver.MustParse("4.0.0")},
+			entry:    ChannelEntry{version: semver.MustParse("4.0.0")},
+			expected: true,
+		},
+		{
+			name:     "Entry is from newer Y-stream",
+			channel:  Channel{yStreamVersion: semver.MustParse("4.0.0")},
+			entry:    ChannelEntry{version: semver.MustParse("4.1.0")},
+			expected: false,
+		},
+		{
+			name:     "Entry is from older Major version",
+			channel:  Channel{yStreamVersion: semver.MustParse("5.0.0")},
+			entry:    ChannelEntry{version: semver.MustParse("4.0.0")},
+			expected: true,
+		},
+		{
+			name:     "Entry is from older Y-stream",
+			channel:  Channel{yStreamVersion: semver.MustParse("4.1.0")},
+			entry:    ChannelEntry{version: semver.MustParse("4.0.6")},
+			expected: true,
+		},
+		{
+			name:     "Entry minor equals channel minor",
+			channel:  Channel{yStreamVersion: semver.MustParse("4.1.0")},
+			entry:    ChannelEntry{version: semver.MustParse("4.1.5")},
+			expected: true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			result := channelShouldHaveEntry(tt.channel, tt.entry)
+			assert.Equal(t, tt.expected, result)
+		})
```

Comment:
Looking again at this, you can save quite a bit on ceremony and reduce reader's mental overhead if you would only set versions in test data and construct all things in the test:
```suggestion
		{
			name:     "Entry minor equals channel minor",
			channelVersion:  "4.1.0",
			entryVersion:    "4.1.5",
			expectedToHave: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			channel := Channel{yStreamVersion: semver.MustParse(tt.channelVersion)}
			entry := ChannelEntry{version: semver.MustParse(tt.entryVersion)}

			result := channelShouldHaveEntry(channel, entry)

			assert.Equal(t, tt.expectedToHave, result)
		})
```